crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -240,4 +243,59 @@ impl GeneratedAcir {
 
         Ok(())
     }
+
+    /// Returns a `Witness` that is constrained to be:
+    /// - `1` if lhs >= rhs
+    /// - `0` otherwise
+    ///
+    /// See [R1CS Workshop - Section 10](https://github.com/mir-protocol/r1cs-workshop/blob/master/workshop.pdf)
+    /// for an explanation.
+    pub(crate) fn more_than_eq_comparison(
+        &mut self,
+        a: &Expression,
+        b: &Expression,
+        max_bits: u32,
+    ) -> Result<Witness, AcirGenError> {
+        // Ensure that 2^{max_bits + 1} is less than the field size
+        //
+        // TODO: perhaps this should be a user error, instead of an assert
```

Comment:
TODO:

The previous SSA pass simply forbidded integers that were more than 127 bits so it made sense to make this a compiler error. This assert should have a message inside of it stating that this is the reason we make it a compiler error and we should also enforce this inside of the range constraint method.

Perhaps we could allow integers of size upto max_bits + 1 and then apply the 127 bit restriction to the places that we know we should not be using them, for example multiplication.

For reference, the 127 is computed by  taking `FieldElement::max_num_bits() / 2`

Reply:
Effectively also tracked by https://github.com/noir-lang/noir/issues/1443 right?

Reply:
Yep resolving as I linked to that issue